builders/reservationBuilder.go

```diff
@@ -0,0 +1,106 @@
+package builders
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"io"
+	"time"
+
+	"github.com/threefoldtech/tfexplorer/models/generated/workloads"
+	"github.com/threefoldtech/tfexplorer/schema"
+	"github.com/threefoldtech/zos/pkg"
+	"github.com/threefoldtech/zos/pkg/crypto"
+)
+
+// ReservationBuilder is a struct that can build reservations
+type ReservationBuilder struct {
+	reservation workloads.Reservation
+}
+
+// NewReservationBuilder creates a new ReservationBuilder
+func NewReservationBuilder() *ReservationBuilder {
+	reservation := workloads.Reservation{}
+	return &ReservationBuilder{
+		reservation: reservation,
+	}
+}
+
+// LoadReservationBuilder loads a reservation builder based on a file path
+func LoadReservationBuilder(reader io.Reader) (*ReservationBuilder, error) {
+	reservation := workloads.Reservation{}
+	err := json.NewDecoder(reader).Decode(&reservation)
+	if err != nil {
+		return &ReservationBuilder{}, err
+	}
+
+	return &ReservationBuilder{
+		reservation: reservation,
+	}, nil
+}
+
+// Save saves the reservation builder to an IO.Writer
+func (r *ReservationBuilder) Save(writer io.Writer) error {
+	err := json.NewEncoder(writer).Encode(r.reservation)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+// Build returns the reservation
+func (r *ReservationBuilder) Build() workloads.Reservation {
+	r.reservation.Epoch = schema.Date{Time: time.Now()}
+	return r.reservation
+}
+
+// WithDuration sets the duration to the reservation
+func (r *ReservationBuilder) WithDuration(duration time.Duration) *ReservationBuilder {
+	timein := time.Now().Local().Add(duration)
+	r.reservation.DataReservation.ExpirationReservation = schema.Date{Time: timein}
+	r.reservation.DataReservation.ExpirationProvisioning = schema.Date{Time: timein}
```

Comment:
You should not set the `ExpirationProvisioning` to the same value as `ExpirationReservation`.
Usually you want to have a relatively small duration for the provision value. This is the time during which the user will have to pay his reservation. The escrow cancel the reservation when this ExpirationProvisioning is reached. This means for user implementing logic on top the can also use this duration as timeout for they "wait" time when deploying.

Let's add an extra WithConfig for ExpirationProvisioning